docs/getting-started/walkthrough.md

```diff
@@ -24,177 +24,411 @@ specific language governing permissions and limitations
-package org.apache.flink.statefun.examples.greeter;
+    egress_message = kafka_egress_record(topic="greetings", key=greet_request.name, value=response)
+    context.pack_and_send_egress("example/greets", egress_message)
+{% endhighlight %}
 
-import org.apache.flink.statefun.sdk.Context;
-import org.apache.flink.statefun.sdk.StatefulFunction;
-import org.apache.flink.statefun.sdk.annotations.Persisted;
-import org.apache.flink.statefun.sdk.state.PersistedValue;
+The state, `seen_count` is always scoped to the current name so it can track each user independently.
 
-public final class GreetFunction implements StatefulFunction {
+## Wiring It All Together
 
-    @Persisted
-    private final PersistedValue<Integer> count = PersistedValue.of("count", Integer.class);
+Stateful Function applications communicate with the Apache Flink runtime using `http`.
+The Python SDK ships with a ``RequestReplyHandler`` that automatically dispatches function calls based on RESTful HTTP ``POSTS``.
+The ``RequestReplyHandler`` may be exposed using any HTTP framework.
 
-    @Override
-    public void invoke(Context context, Object input) {
-        GreetRequest greetMessage = (GreetRequest) input;
+One popular Python web framework is [Flask](https://palletsprojects.com/p/flask/).
+It can be used to quickly and easily expose an application to the Apache Flink runtime.
 
-        GreetResponse response = computePersonalizedGreeting(greetMessage);
+{% highlight python %}
+from statefun import StatefulFunctions
+from statefun import RequestReplyHandler
 
-        context.send(GreetingConstants.GREETING_EGRESS_ID, response);
-    }
+functions = StatefulFunctions()
 
-    private GreetResponse computePersonalizedGreeting(GreetRequest greetMessage) {
-        final String name = greetMessage.getWho();
-        final int seen = count.getOrDefault(0);
-        count.set(seen + 1);
+@functions.bind("walkthrough/greeter")
+def greeter(context, message: GreetRequest):
+    pass
 
-        String greeting = greetText(name, seen);
+handler = RequestReplyHandler(functions)
 
-        return GreetResponse.newBuilder()
-            .setWho(name)
-            .setGreeting(greeting)
-            .build();
-    }
-}
+# Serve the endpoint
+
+from flask import request
+from flask import make_response
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route('/statefun', methods=['POST'])
+def handle():
+    response_data = handler(request.data)
+    response = make_response(response_data)
+    response.headers.set('Content-Type', 'application/octet-stream')
+    return response
+
+
+if __name__ == "__main__":
+    app.run()
 {% endhighlight %}
 
-Each time a message is processed, the function computes a personalized message for that user.
-It reads and updates the number of times that user has been seen and sends a greeting to the egress.
+## Configuring for Runtime
+
+The Apache Flink runtime makes requests to the greeter function by making `http` calls to the `Flask` server.
```

Comment:
Just a thought: Not sure how we are positioning StateFun, but this sounds like something coming directly from Flink?
Maybe we could somehow phrase this to come from the "server side" of statefun running in Flink?

If we are doing a good job, we might have users who are not even really aware that this is powered by Flink.

Reply:
I'll find a way to rephrase it, you're right, you don't need to understand Flink for this getting started.

That said, I don't think you could go to production without understanding the project's relationship to Flink. We may want to follow up with a dedicated page explaining how these projects are related.